File: CSharpCodeBase/aiactions/ai_selectenemy.lua
local Class = require 'utils.hump.class'
local AINode = require 'utils.ailib.ainode'

-- Select enemy using ai vision
local AI_SelectEnemy = Class({__includes=AINode, init=function(self, entity)
  AINode.init(self, entity)
end})

function AI_SelectEnemy:Visit()
  if self.status == NODE_READY then
    self.parent.selected_target = self.entity:GetPriorityTarget()
    self.status = self.parent.selected_target ~= nil and NODE_SUCCESS or NODE_FAILURE
  end
  return self.status
end

return AI_SelectEnemy

File: CSharpCodeBase/aiactions/ai_selectenemy.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class ai_selectenemy :ainode{
 // Select enemy using ai vision
 public void init(self, entity);
   AINode.init(self, entity);
 }})
 public void Visit(){
   if(this.status == NODE_READY  ){ 
     this.parent.selected_target = this.entity:GetPriorityTarget();
     this.status = this.parent.selected_target != null  &&  NODE_SUCCESS  ||  NODE_FAILURE;
   }
   return this.status;
 }
 }}